src/resolve/connectionResolver.ts

```diff
@@ -98,9 +98,24 @@ export class ConnectionResolver {
   private async listMembers(query: ListMetadataQuery): Promise<FileProperties[]> {
     let members: FileProperties[];
 
+    const pollingOptions: PollingClient.Options = {
+      frequency: Duration.milliseconds(1000),
+      timeout: Duration.minutes(3),
+      poll: async (): Promise<StatusResult> => {
+        const res = ensureArray(await this.connection.metadata.list(query));
+        return { completed: true, payload: res };
+      },
+    };
+
+    const pollingClient = await PollingClient.create(pollingOptions);
+
     try {
-      members = ensureArray((await this.connection.metadata.list(query)) as FileProperties[]);
+      members = await pollingClient.subscribe();
     } catch (error) {
+      // throw error if PollingClient timed out.
+      if (error instanceof NotRetryableError) {
+        throw NotRetryableError;
```

Comment:
this catch block ignores error on purpose (see lines below) so we need to check if the error comes from the polling client.